tests/ttnn/unit_tests/operations/pool/test_avgpool2d.py

```diff
@@ -0,0 +1,137 @@
+# SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+
+import torch
+import ttnn
+import pytest
+
+from tests.ttnn.utils_for_testing import assert_with_pcc
+
+
+def run_avg_pool(
+    device, input_shape, kernel_size, stride, padding, dilation, ceil_mode, shard_scheme, memory_config=None
+):
+    ## Test setup for both.
+    in_n, in_c, in_h, in_w = input_shape
```

Comment:
you can take a look how we cache torch input tensors so we speed up test execution, but not generating new random torch tensors for 
test variants that have same input dims